platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultDependencyConstraintFactory.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2011 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts;
+
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.DependencyConstraint;
+import org.gradle.api.artifacts.MinimalExternalModuleDependency;
+import org.gradle.api.internal.artifacts.dependencies.DefaultDependencyConstraint;
+import org.gradle.api.internal.artifacts.dsl.dependencies.DependencyConstraintFactoryInternal;
+import org.gradle.api.internal.attributes.ImmutableAttributesFactory;
+import org.gradle.api.internal.notations.DependencyConstraintNotationParser;
+import org.gradle.internal.reflect.Instantiator;
+
+import javax.annotation.Nullable;
+
+
+public class DefaultDependencyConstraintFactory implements DependencyConstraintFactoryInternal {
```

Comment:
Why is this separate from `DependencyFactory`? How often does somebody want dependencies but not constraints, or the other way around? I think in general people will want to create both or either and having to inject a separate type makes it just a little bit more annoying to do that.

Reply:
I think it makes sense to keep them distinct.  Platforms are (should be) constraints without dependencies, so it's not like you always want both.